servicetalk-http-netty/src/test/java/io/servicetalk/http/api/BaseContentCodingTest.java

```diff
@@ -0,0 +1,245 @@
+                when(H2, Server.ID_DEFLATE, Client.DEFLATE_ID, Request.DEFLATE, SHOULD_PASS),
+                when(H1, Server.ID_DEFLATE, Client.DEFAULT, Request.ID, SHOULD_PASS),
+                when(H2, Server.ID_DEFLATE, Client.DEFAULT, Request.ID, SHOULD_PASS),
+                when(H1, Server.GZIP_ONLY, Client.ID_ONLY, Request.ID, SHOULD_PASS),
+                when(H2, Server.GZIP_ONLY, Client.ID_ONLY, Request.ID, SHOULD_PASS),
+                when(H1, Server.GZIP_ONLY, Client.GZIP_ID, Request.ID, SHOULD_PASS),
+                when(H2, Server.GZIP_ONLY, Client.GZIP_ID, Request.ID, SHOULD_PASS),
+                when(H1, Server.GZIP_ONLY, Client.GZIP_ID, Request.ID, SHOULD_PASS),
+                when(H2, Server.GZIP_ONLY, Client.GZIP_ID, Request.ID, SHOULD_PASS),
+                when(H1, Server.GZIP_ONLY, Client.GZIP_ID, Request.GZIP, SHOULD_PASS),
+                when(H2, Server.GZIP_ONLY, Client.GZIP_ID, Request.GZIP, SHOULD_PASS),
+                when(H1, Server.DEFAULT, Client.GZIP_ID, Request.GZIP, SHOULD_FAIL),
+                when(H2, Server.DEFAULT, Client.GZIP_ID, Request.GZIP, SHOULD_FAIL),
+                when(H1, Server.DEFAULT, Client.GZIP_DEFLATE_ID, Request.DEFLATE, SHOULD_FAIL),
+                when(H2, Server.DEFAULT, Client.GZIP_DEFLATE_ID, Request.DEFLATE, SHOULD_FAIL),
+                when(H1, Server.DEFAULT, Client.GZIP_ID, Request.ID, SHOULD_PASS),
+                when(H2, Server.DEFAULT, Client.GZIP_ID, Request.ID, SHOULD_PASS),
+        };
+    }
+
+    @Test
+    public void testCompatibility() throws Exception {
+        if (expectedSuccess) {
+            assertSuccessful(scenario.requestEncoding);
+        } else {
+            assertNotSupported(scenario.requestEncoding);
+        }
+    }
+
+    protected abstract void assertSuccessful(ContentCodec requestEncoding) throws Exception;
+
+    protected abstract void assertNotSupported(ContentCodec requestEncoding) throws Exception;
+
+    protected static byte[] payload(byte b) {
+        byte[] payload = new byte[PAYLOAD_SIZE];
+        Arrays.fill(payload, b);
+        return payload;
+    }
+
+    protected static String payloadAsString(byte b) {
+        return new String(payload(b), StandardCharsets.US_ASCII);
+    }
+
+    protected enum Server {
+        DEFAULT(emptyList()),
+        GZIP_ONLY(singletonList(gzipDefault())),
+        GZIP_ID(asList(gzipDefault(), identity())),
+        GZIP_DEFLATE_ID(asList(gzipDefault(), deflateDefault(), identity())),
+        ID_GZIP(asList(identity(), gzipDefault())),
+        ID_DEFLATE(asList(identity(), deflateDefault())),
+        ID_GZIP_DEFLATE(asList(identity(), gzipDefault(), deflateDefault())),
+        DEFLATE_ONLY(singletonList(deflateDefault())),
+        DEFLATE_ID(asList(deflateDefault(), identity()));
+
+        List<ContentCodec> list;
+
+        Server(List<ContentCodec> list) {
+            this.list = list;
+        }
+    }
+
+    protected enum Client {
```

Comment:
`Client` and `Server` enums are identical. Can we rename it to the generic name and reuse?

Reply:
Used different names to make the `when`read better in English.
"when(H1, Server.DEFAULT, Client.DEFAULT, Request.ID, SHOULD_PASS)"